x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -385,39 +390,8 @@ public PhysicalPlan physicalPlan(LogicalPlan logicalPlan) {
 
     public PhysicalPlan optimizedPhysicalPlan(LogicalPlan logicalPlan) {
         var plan = physicalPlanOptimizer.optimize(physicalPlan(logicalPlan));
+        planningMetricsManager.publish(planningMetrics);
```

Comment:
Is this really the ok place to add this `publish`? This method is likely to be called from the recent code that Nik added (multi-phased approach for `inlinestats`) multiple times.

Reply:
Good point, let me review it

Reply:
Moved it at `PlanExecutor` level, so that we can also track success/failure